x-pack/plugins/infra/public/components/source_configuration/ml_configuration_panel.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiTitle } from '@elastic/eui';
+import { EuiSpacer } from '@elastic/eui';
+import { EuiFormRow } from '@elastic/eui';
+import { EuiRange } from '@elastic/eui';
+import { EuiDescribedFormGroup } from '@elastic/eui';
+import { EuiForm } from '@elastic/eui';
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import { InputRangeFieldProps } from './input_fields';
+
+interface MLConfigurationPanelProps {
+  isLoading: boolean;
+  readOnly: boolean;
+  anomalyThresholdFieldProps: InputRangeFieldProps;
+}
+
+export const MLConfigurationPanel = ({
+  isLoading,
+  readOnly,
+  anomalyThresholdFieldProps,
+}: MLConfigurationPanelProps) => {
+  return (
+    <EuiForm>
+      <EuiTitle size="s">
+        <h3>
+          <FormattedMessage
+            id="xpack.infra.sourceConfiguration.mlSectionTitle"
+            defaultMessage="Machine Learning"
+          />
+        </h3>
+      </EuiTitle>
+      <EuiSpacer size="m" />
+      <EuiDescribedFormGroup
+        title={
+          <h4>
+            <FormattedMessage
+              id="xpack.infra.sourceConfiguration.anomalyThresholdTitle"
+              defaultMessage="Anomaly Severity Threshold"
+            />
+          </h4>
+        }
+        description={
+          <FormattedMessage
+            id="xpack.infra.sourceConfiguration.anomalyThresholdDescription"
+            defaultMessage="Sets the minimum severity score required to show anomalies in the Metrics application."
+          />
+        }
+      >
+        <EuiFormRow
+          error={anomalyThresholdFieldProps.error}
+          fullWidth
+          isInvalid={anomalyThresholdFieldProps.isInvalid}
+          label={
+            <FormattedMessage
+              id="xpack.infra.sourceConfiguration.anomalyThresholdLabel"
+              defaultMessage="Minimum severity score"
+            />
+          }
+        >
+          <EuiRange
+            min={0}
+            max={100}
+            data-test-subj="anomalyThresholdInput"
+            showInput
+            showTicks
+            showRange
+            tickInterval={20}
```

Comment:
Is it possible to get this to line up with the ML plugin's defined severity levels? I believe they go in intervals of 25, but the lowest level is 3, so it's kind of weird. See https://github.com/elastic/kibana/blob/master/x-pack/plugins/ml/common/constants/anomalies.ts

Maybe we could achieve this just by setting `tickInterval` to 25, but ideally I'd like us to be able to import these constants in case they're ever changed. Not sure if EuiRange supports that kind of thing.

Reply:
If it's a problem to set the `min` to 3 (i.e. you can start at 3 but then with a `tickInterval` of 25 it'd jump to 27, 52, 77), it's probably fine to set it to `0`. My only concern is if there's a situation where the ML plugin records an anomaly with a score below 3 in the database, and expects all UI components to just filter that out, but I think that's unlikely.

Reply:
I'd also thought of this.   There isn't any information here for the user to understand what a number from 0-100 means.  Currently they can select any number from 0-100.  @Zacqary Are you suggesting they only be allowed to select one of the security level numbers?   The `EuiRange` allows adding some colors to certain ranges but there doesn't seem to be a way to limit them from selecting only certain numbers if it doesn't fit within the `step`, so 3 in this case, if `step` was 25.   We could have to use the color options available `[ "primary", "success", "warning", "danger" ]`.

Reply:
Yeah I think limiting the selection to correspond to the exact threshold levels is a good idea. I think we can use `<EuiRangeTicks>` to label them with words instead of numbers. Probably don't need the colors, in that case, although it looks like the ML team has put in a feature request for this exact use case: https://github.com/elastic/eui/issues/4467

Conveniently, that linked issue includes a screenshot of what this slider should probably look like.

Reply:
I don't think we can limit the exact numbers the user is able to select using the EuiRange component.  I'm not really sure as I can see the user wanting to be able to select any number they wish and having some precision.  I'm thinking we should probably just merge this for now though and improve on the UX later.  @katefarrar thoughts?

Reply:
Thanks for all the context @Zacqary & @neptunian. I agree we should merge this for now and we can explore UI improvements in the future. Also, I'd like to see what the EUI team comes up with here before we do any custom work on our end.